src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -1139,49 +1354,367 @@ public static Vector512<float> Invoke(Vector512<float> x, Vector512<float> y)
             public static Vector512<float> Invoke(Vector512<float> x, Vector512<float> y) => x * y;
 #endif
 
+            public static float Invoke(Vector128<float> x) => HorizontalAggregate<MultiplyOperator>(x);
+            public static float Invoke(Vector256<float> x) => HorizontalAggregate<MultiplyOperator>(x);
+#if NET8_0_OR_GREATER
+            public static float Invoke(Vector512<float> x) => HorizontalAggregate<MultiplyOperator>(x);
+#endif
+        }
+
+        private readonly struct DivideOperator : IBinaryOperator
+        {
+            public static float Invoke(float x, float y) => x / y;
+            public static Vector128<float> Invoke(Vector128<float> x, Vector128<float> y) => x / y;
+            public static Vector256<float> Invoke(Vector256<float> x, Vector256<float> y) => x / y;
+#if NET8_0_OR_GREATER
+            public static Vector512<float> Invoke(Vector512<float> x, Vector512<float> y) => x / y;
+#endif
+        }
+
+        private readonly struct MaxOperator : IBinaryOperator
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static float Invoke(float result, float current) =>
+                current == result ?
+                    (IsNegative(result) ? current : result) :
+                    (current > result ? current : result);
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector128<float> Invoke(Vector128<float> result, Vector128<float> current) =>
+                Vector128.ConditionalSelect(Vector128.Equals(result, current),
+                    Vector128.ConditionalSelect(IsNegative(result), current, result),
+                    Vector128.Max(result, current));
```

Comment:
nit: afair, should be enough to use just `Vector128.Max` on arm as far as I remember, it handles -0.0/0.0

Reply:
Right. `Arm64` has an `IEEE 754` compliant `min`, `max`, `minMagnitude`, and `maxMagnitude` instructions directly available.

x64 does not, but you can do optimized alternative sequences for some scenarios.